### PR TITLE
Admin Section Improved

### LINKED ISSUE #21: Admin Section Improve
Admin Section is Not showing All Right Data

### PR DESCRIPTION
**Pull request by rkishan516**
Admin section is now showing correct Data
Fixed #21 
screenshot from 2018-12-23 12-16-41
Admin Can See Online Users
screenshot from 2018-12-25 17-12-30

### CODE REVIEW

**Comment by anuragteapot** [comment]
@rkishan516 Check error.log

**Comment by anuragteapot** [comment]
@rkishan516 Please check this branch. https://github.com/Anu1601CS/iitpConnect/tree/deploy

I got some issue with login track please make the changes according to this branch only for login track.

**Comment by rkishan516** [comment]
@Anu1601CS But this branch doesn't contain admin section than how can it be getting error

**Comment by anuragteapot** [comment]
`@Anu1601CS But this branch doesn't contain admin section than how can it be getting error`

@rkishan516 I am talking about `isOnline` and `goOffline` functions. Don't see admin.   See things only related to login traking.

**Comment by anuragteapot** [comment]
This issue is not related to this PR

File: admin/views/Home.php [hunk 1]
```diff
@@ -310,15 +317,15 @@
...
                                                 $holidayList = Admin::fetchHolidays();
-                                                while($holidayList = mysqli_fetch_array($holidayList))
+                                                while($row = mysqli_fetch_assoc($holidayList))
                                                 {
-                                                    $id = $holidayList['id'];
-                                                    $name = $holidayList['name'];
-                                                    $date = $holidayList['holidayDate'];
-                                                    $type = $holidayList['type'];
-                                                    echo "<tr class='odd gradeX'> <td> $id </td><td> $name </td><td> $data </td><td> $type </td><td> $type </td>";
+                                                    $id = $row['id'];
```
Comment (anuragteapot) [inline admin/views/Home.php]: Why we need a new variable.?

File: admin/views/Home.php [hunk 2]
```diff
@@ -247,16 +252,18 @@
...
-                                                    $title = $postData['title'];
-                                                    $postId = $postData['postId'];
-                                                    $likes = $postData['Likes'];
-                                                    $reports = $postData['reports'];
-
-                                                    echo "<tr class='odd gradeX'> <td> $postId </td><td> $title </td><td> $likes </td><td> $reports </td><td> $username </td>";
+                                                    $postData = Admin::fetchPostsDetails($userid['id']);
+                                                    while($row = mysqli_fetch_assoc($postData))
+                                                    {
+                                                        $username =  $row['username'];
```
Comment (anuragteapot) [inline admin/views/Home.php]: Same here

File: admin/views/Home.php [hunk 3]
```diff
@@ -183,15 +183,20 @@
...
-                                                    $userData = Admin::fetchUserPosts($userid);
-                                                    $username =  $userData['username'];
-                                                    $totalPost = $userData['totalPosts'];
-                                                    $followers = $userData['followers'];
-                                                    $totalLikes = $userData['totalLikes'];
+                                                    $userData = Admin::fetchUserPosts($userid['id']);
+                                                    while($row = mysqli_fetch_assoc($userData))
+                                                    {
+                                                        $id = $userid['id'];
+                                                        $username =  $row['username'];
```
Comment (anuragteapot) [inline admin/views/Home.php]: No need of new variable

File: admin/views/Home.php [hunk 4]
```diff
@@ -374,12 +381,12 @@
...
                                                 $feedbackList = Admin::fetchFeedback();
-                                                while($feedbackList = mysqli_fetch_array($feedbackList))
+                                                while($row = mysqli_fetch_array($feedbackList))
                                                 {
-                                                    $id = $feedbackList['id'];
-                                                    $name = $feedbackList['name'];
-                                                    $email = $feedbackList['email'];
-                                                    $message = $feedbackList['message'];
+                                                    $id = $row['id'];
+                                                    $name = $row['name'];
```
Comment (anuragteapot) [inline admin/views/Home.php]: Same here

File: libraries/classes/Admin.php [hunk 5]
```diff
@@ -15,10 +15,20 @@ public static function getAllUserDetails()
       $db = new Factory;
       $mysql = $db->getDBO();
   
-      $sql  = "SELECT sum(po.status) as totalPosts,sum(us.activation) as totalUsers, from posts po INNER JOIN users us ON po.uid = us.id WHERE 1"; 
-  
+      $sql  = "SELECT sum(us.activation) as totalUsers from users us";
```
Comment (anuragteapot) [inline libraries/classes/Admin.php:18]: Learn about joins to make this one query. 
https://www.tutorialspoint.com/sql/sql-using-joins.htm

**Review by anuragteapot — APPROVED** [review]
Good Work

**Pull request merged**